doc/book/quick-start-skeleton.md

```diff
@@ -0,0 +1,259 @@
+            'allowed_methods' => ['GET'],
+        ],
+    ],
+];
+```
+
+Once you've added the above entry, give it a try by going to each of the
+following URIs:
+
+- http://localhost:8080/hello
+- http://localhost:8080/hello?target=ME
+
+You should see the message change as you go between the two URIs!
+
+### Using templates
+
+You likely don't want to hardcode HTML into your middleware; so, let's use
+templates. This particular exercise assumes you chose to use the
+[Plates](http://platesphp.com) integration.
+
+Templates are installed under the `templates/` subdirectory. By default, we also
+register the template namespace `app` to correspond with the `templates/app`
+subdirectory. Create the file `templates/app/hello-world.phtml` with the
+following contents:
+
+```php
+<?php $this->layout('layout::default', ['title' => 'Greetings']) ?>
+
+<h2>Hello, <?= $this->e($target) ?></h2>
+```
+
+Now that we have a template, we need to:
+
+- Inject a renderer into our action class.
+- Use the renderer to render the contents.
+
+Replace your `src/Action/HelloAction.php` file with the following contents:
+
+```php
+<?php
+namespace App\Action;
+
+use Zend\Diactoros\Response\HtmlResponse;
+use Zend\Expressive\Template\TemplateRendererInterface;
+
+class HelloAction
+{
+    private $renderer;
+
+    public function __construct(TemplateRendererInterface $renderer)
+    {
+        $this->renderer = $renderer;
+    }
+
+    public function __invoke($request, $response, callable $next)
+    {
+        $query  = $request->getQueryParams();
+        $target = isset($query['target']) ? $query['target'] : 'World';
+
+        return new HtmlResponse(
+            $this->renderer->render('app::hello-world', ['target' => $target])
+        );
+    }
+}
+```
+
+The above modifies the class to accept a renderer to the constructor, and then
+calls on it to render a template. A few things to note:
+
+- We no longer need to escape our target; the template takes care of that for us.
+- We're using a specific response type here, from
+  [Diactoros](https://github.com/zendframework/zend-diactoros), which is the
+  default PSR-7 implementation Expressive uses. This response type simplifies
+  our response creation.
+
+How does the template renderer get into the action, however? The answer is
+dependency injection.
+
+Let's create a factory. Create the file `src/Action/HelloActionFactory.php` with
+the following contents:
+
+```php
+<?php
+namespace App\Action;
+
+use Interop\Container\ContainerInterface;
+use Zend\Expressive\Template\TemplateRendererInterface;
+
+class HelloActionFactory
+{
+    public function __invoke(ContainerInterface $container)
+    {
+        return new HelloAction(
+            $container->get(TemplateRendererInterface::class)
+        );
+    }
+}
+```
+
+With that in place, we'll now update our configuration. Open the file
+`config/autoload/dependencies.global.php`; it should have a structure similar to
```

Comment:
No. They should be on the dependencies.global.php file.
On Oct 16, 2015 9:39 PM, "Daniel Gimenes" <notifications@github.com> wrote:

> In doc/book/quick-start-skeleton.md
> <https://github.com/zendframework/zend-expressive/pull/158#discussion_r42304685>
> :
>
> > +use Interop\Container\ContainerInterface;
> > +use Zend\Expressive\Template\TemplateRendererInterface;
> > +
> > +class HelloActionFactory
> > +{
> > +    public function __invoke(ContainerInterface $container)
> > +    {
> > +        return new HelloAction(
> > +            $container->get(TemplateRendererInterface::class)
> > +        );
> > +    }
> > +}
> > +```
> > +
> > +With that in place, we'll now update our configuration. Open the file
> > +`config/autoload/dependencies.global.php`; it should have a structure similar to
>
> But the middleware factories should not be declared in routes.global.php?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-expressive/pull/158/files#r42304685>
> .
>